### PR TITLE
Upgrade `WorkManager` to version 2.2.0

### PR DESCRIPTION
**Pull request by Dexterp37**
This was a bit less trivial than expected: the new version required a version upgrade of some other dependencies as well, so I ended up upgrading them all. We are now matching the [same version that Android Components is using](https://github.com/mozilla-mobile/android-components/blob/master/buildSrc/src/main/java/Dependencies.kt).

This allowed to simplify the code in a few places: see the 'BONUS' commits.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -11,28 +11,29 @@ buildscript {
     // changing them. Please note that, for using in Android-Components, the
     // versions below must match the ones in that repository.
     ext.versions = [
-        android_components: '5.0.0',
+        android_components: '15.0.0',
```
Comment (travis79) [inline build.gradle]: Whoa! That was a big jump.

File: build.gradle [hunk 2]
```diff
@@ -11,28 +11,29 @@ buildscript {
...
+        androidx_annotation: '1.1.0',
+        androidx_appcompat: '1.1.0',
+        androidx_browser: '1.2.0-alpha07',
+        androidx_core: '1.1.0',
         androidx_espresso: '3.2.0',
         androidx_junit: '1.1.1',
+        androidx_lifecycle_extensions: '2.1.0',
         androidx_test: '1.2.0',
-        androidx_work: '2.0.1',
+        androidx_work: '2.2.0',
```
Comment (travis79) [inline build.gradle]: Welcome to WorkManager 2.2, where we fixed all the old bugs and introduced new ones!

File: build.gradle (hunk 1 above)
Comment (Dexterp37) [reply build.gradle]: Ehehhe, yeah :) We're just using this for tests, so it should not be too pressing to update it :D

File: build.gradle (hunk 2 above)
Comment (Dexterp37) [reply build.gradle]: Ahahaha, right :D Let's hope more was fixed than broken :P

**Comment by Dexterp37** [comment]
> 
> 
> This touched a lot more than I expected but it looks good! I had one question that came up while looking through this and that was where we use a mocked/spy context, I assume that this doesn't affect the WorkManager?

Nope, it doesn't seem to affect it!

**Comment by travis79** [comment]
> > This touched a lot more than I expected but it looks good! I had one question that came up while looking through this and that was where we use a mocked/spy context, I assume that this doesn't affect the WorkManager?
> 
> Nope, it doesn't seem to affect it!

Sweet!

**Pull request merged**